src/test/java/redis/clients/jedis/tests/ConnectionTest.java

```diff
@@ -4,51 +4,60 @@
 import static org.junit.Assert.fail;
 
 import org.junit.After;
-import org.junit.Before;
 import org.junit.Test;
 
 import redis.clients.jedis.Connection;
+import redis.clients.jedis.Protocol;
 import redis.clients.jedis.Protocol.Command;
 import redis.clients.jedis.commands.ProtocolCommand;
 import redis.clients.jedis.exceptions.JedisConnectionException;
 
 public class ConnectionTest {
   private Connection client;
 
-  @Before
-  public void setUp() throws Exception {
-    client = new Connection();
-  }
-
   @After
   public void tearDown() throws Exception {
-    client.close();
+    if (client != null) {
```

Comment:
@sazzad16 why do you still need this `@After` if the connection is not created in the `@Before` any more?

Reply:
@gkorland This test class mostly consists of constructor tests. So connections are supposed to be created in each test methods. I'm closing those objects here in `@After`.